### ISSUE TITLE
CLLE language server output issue: "Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')"

### ISSUE THREAD

**Issue by MarcoDeSenas** [halcyon-tech/vscode-ibmi#1495]
Hello, I get this error while editing a clle IFS source file.

[Error - 2:36:37 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 
[Error - 2:36:38 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 
[Error - 2:36:39 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 
[Error - 2:36:44 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 
[Error - 2:36:46 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 
[Error - 2:37:02 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 
[Error - 2:37:05 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 
[Error - 2:37:13 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 
[Error - 2:37:13 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 
[Error - 2:37:13 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 
[Error - 2:37:14 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 
[Error - 2:37:14 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 
[Error - 2:37:14 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 
[Error - 2:37:14 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read properties of undefined (reading 'map')
  Code: -32603 

<hr />

|Context|Version|
|-|-|
|Code for IBM i version|2.1.0|
|Visual Studio Code version|1.81.1|
|Operating System|win32_x64|

<details>
<summary>Active extensions</summary>

```
CL (vscode-clle): 1.1.1
COBOL (cobol): 9.7.23
Code Coverage for IBM i (code-coverage-ibmi): 0.1.7
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.3.1
Db2 for IBM i (vscode-db2i): 0.4.0
Emmet (emmet): 1.0.0
Error Lens (errorlens): 3.13.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
GitHub Authentication (github-authentication): 0.0.2
IBM i Notebooks (vscode-ibmi-notebooks): 0.0.6
JSON Language Features (json-language-features): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
Microsoft Account (microsoft-authentication): 0.0.1
Node Debug Auto-attach (debug-auto-launch): 1.0.0
Overtype (overtype): 0.5.0
TODO Highlight (vscode-todo-highlight): 1.0.5
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
WSL: Recommender (remote-wsl-recommender): 0.0.19
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|V7R5M0|
|Tech Refresh|n/a|
|CCSID|297|
|SQL|Enabled
|Source dates|Disabled

### Enabled features

|/QOpenSys/pkgs/bin|/usr/bin|/QSYS.lib/DIMARCOB.lib|/QSYS.LIB|/QIBM/ProdData/IBMiDebugService/bin|
|-|-|-|-|-|
|bash|attr|GENCMDXML.PGM|QZDFMDB2.PGM|startDebugService.sh|
|chsh|iconv|GETNEWLIBL.PGM|||
|git|setccsid||||
|grep|||||
|ls|||||
|md5sum|||||
|sort|||||
|stat|||||
|tar|||||
|tn5250|||||
</details>

<details>
<summary>Shell env</summary>

```bash
BUILDLIB=DIMARCO1
CURLIB=DIMARCO1
HOME=/home/DIMARCO
LIBLS=DIMARCOB GAMES400 DIMARCO2 QTEMP QGPL
LOGIN=dimarco
LOGNAME=dimarco
MAIL=/var/spool/mail/dimarco
OLDPWD=/home/DIMARCO
PASE_USRGRP_LIMITED=N
PATH=/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin
PWD=/home/DIMARCO
SHELL=/QOpenSys/pkgs/bin/bash
SHLVL=1
SSH_CLIENT=88.170.41.66 36799 2222
SSH_CONNECTION=88.170.41.66 36799 185.113.5.134 2222
TZ=<UTC>0
USER=dimarco
USERNAME=DIMARCO
_=/QOpenSys/usr/bin/env
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "£à$"
}
```
</details>

<details>
<summary>Errors</summary>

```json
[
  {
    "command": "/usr/bin/attr \"/tmp/S         \" CCSID",
    "code": 1,
    "stderr": "attr: 001-2248 Error found getting attributes for file /tmp/S         .  No such path or directory.",
    "cwd": "/home/DIMARCO"
  },
  {
    "command": "/usr/bin/attr \"/tmp/D         \" CCSID",
    "code": 1,
    "stderr": "attr: 001-2248 Error found getting attributes for file /tmp/D         .  No such path or directory.",
    "cwd": "/home/DIMARCO"
  },
  {
    "command": "/usr/bin/attr \"/tmp/W         \" CCSID",
    "code": 1,
    "stderr": "attr: 001-2248 Error found getting attributes for file /tmp/W         .  No such path or directory.",
    "cwd": "/home/DIMARCO"
  }
]
```
</details>

**Comment by worksofliam** [comment]
@MarcoDeSenas Any chance you can share the source with us that was causing the issue?

**Comment by MarcoDeSenas** [comment]
@worksofliam 
Just got one when typing dmpclpgm in the source below.


/*--------------------------------------------------------------------------*/
/*                                                                          */
/* Validity checker program for OBJCRT command.                             */
/*                                                                          */
/* Dates: 2023/08/22 Creation                                               */
/*                                                                          */
/*--------------------------------------------------------------------------*/

PGM PARM(&TYPE &OBJECT &SOURCE)

DCL VAR(&TYPE) TYPE(*CHAR) LEN(7)
DCL VAR(&OBJECT) TYPE(*CHAR) LEN(10)
DCL VAR(&SOURCE) TYPE(*CHAR) LEN(512)

DCL VAR(&PARAM1) TYPE(*CHAR) LEN(10) VALUE('TYPE')
DCL VAR(&PARAM2) TYPE(*CHAR) LEN(10) VALUE('OBJECT')
DCL VAR(&PARAM3) TYPE(*CHAR) LEN(10) VALUE('SOURCE')

INCLUDE SRCSTMF('../../common/includes/inc_variables_declare.clle')

/*--------------------------------------------------------------------------*/
/*                                                                          */
/* We cannot include the standard error handling routine because system     */
/* programs which run the validity checker expect only CPF0002 escape       */
/* when they call it.                                                       */
/* So, if an unexpected error occurs we handle it just like if it was an    */
/* error from one of the checks. And a specific error handling routine      */
/* exists for validity checker programs.                                    */
/*                                                                          */
/*--------------------------------------------------------------------------*/

INCLUDE +
    SRCSTMF('../../common/includes/inc_errorhandling_forchecker_declare.clle')

MONMSG MSGID(CPF0000) EXEC(GOTO CMDLBL(ERROR))

/*--------------------------------------------------------------------------*/
/*                                                                          */
/* Some initializations...                                                  */
/*                                                                          */
/*--------------------------------------------------------------------------*/

INCLUDE SRCSTMF('../../common/includes/inc_variables_init.clle')
CHGVAR VAR(&ERROR) VALUE(&FALSE)
CHGVAR VAR(&ERRORPARAM) VALUE(&FALSE)

/*--------------------------------------------------------------------------*/
/*                                                                          */
/* Checkings are the following:                                             */
/*                                                                          */
/* TYPE must be *QMQRY, *QMFORM, *DSPF or *PRTF                             */
/* OBJECT must not be empty.                                                */
/* SOURCE must not be empty.                                                */
/*                                                                          */
/*--------------------------------------------------------------------------*/

IF COND(&TYPE *NE '*QMQRY' *AND &TYPE *NE '*QMFORM' +
        *AND &TYPE *NE '*DSPF' *AND &TYPE *NE '*PRTF') +
        THEN(DO)
    CHGVAR VAR(&ERRORPARAM) VALUE(&TRUE)
    CHGVAR VAR(&MSGDTA) VALUE('0000')
    CHGVAR VAR(%SST(&MSGDTA 5 10)) VALUE(&PARAM1)
    CHGVAR VAR(%SST(&MSGDTA 15 10)) VALUE(&TYPE)
    SNDPGMMSG MSGID(CPD0084) MSGF(QCPFMSG) MSGTYPE(*DIAG) MSGDTA(&MSGDTA)
ENDDO

IF COND(&OBJECT *EQ &BLANK) THEN(DO)
    CHGVAR VAR(&ERRORPARAM) VALUE(&TRUE)
    CHGVAR VAR(&MSGDTA) VALUE('0000')
    CHGVAR VAR(&MSGDTA) VALUE(&MSGDTA *TCAT &PARAM2)
    SNDPGMMSG MSGID(CPD0071) MSGF(QCPFMSG) MSGTYPE(*DIAG) MSGDTA(&MSGDTA)
ENDDO

IF COND(&SOURCE *EQ &BLANK) THEN(DO)
    CHGVAR VAR(&ERRORPARAM) VALUE(&TRUE)
    CHGVAR VAR(&MSGDTA) VALUE('0000')
    CHGVAR VAR(&MSGDTA) VALUE(&MSGDTA *TCAT &PARAM3)
    SNDPGMMSG MSGID(CPD0071) MSGF(QCPFMSG) MSGTYPE(*DIAG) MSGDTA(&MSGDTA)
ENDDO
**dmpclpgm**
/*--------------------------------------------------------------------------*/
/*                                                                          */
/* Error hangling.                                                          */
/*                                                                          */
/*--------------------------------------------------------------------------*/

INCLUDE +
    SRCSTMF('../../common/includes/inc_errorhandling_forchecker_routine.clle')

ENDPGM

**Comment by worksofliam** [comment]
@MarcoDeSenas Thanks, I've found the issue. It's due to the command having no parameters! A fix will be out today.

**Comment by worksofliam** [comment]
This is fixed in vscode-clle 1.1.2. For the future, CL issues go in the [vscode-clle](https://github.com/IBM/vscode-clle/) repo.
